### ISSUE TITLE
Can't install inside a docker container

### ISSUE THREAD

**Issue by simaiden** [NVIDIA-AI-IOT/torch2trt#483]
Hi, I'm trying to install torch2trt when building  docker image in a jetson xavier nx. I'm using as base image a nvidia container with tensorrt and pytorch installed, but the next error appears:

```
Traceback (most recent call last):
  File "setup.py", line 2, in <module>
    import torch
  File "/usr/local/lib/python3.6/dist-packages/torch/__init__.py", line 188, in <module>
    _load_global_deps()
  File "/usr/local/lib/python3.6/dist-packages/torch/__init__.py", line 141, in _load_global_deps
    ctypes.CDLL(lib_path, mode=ctypes.RTLD_GLOBAL)
  File "/usr/lib/python3.6/ctypes/__init__.py", line 348, in __init__
    self._handle = _dlopen(self._name, mode)
OSError: libcurand.so.10: cannot open shared object file: No such file or directory

```

My dockerfile is:

```
FROM nvcr.io/nvidia/l4t-pytorch:r32.4.4-pth1.6-py3

RUN apt-get update && apt-get install -y \
        python3-opencv ca-certificates python3-dev git wget sudo  \
        cmake ninja-build && \
  rm -rf /var/lib/apt/lists/*
#RUN ln -sv /usr/bin/python3 /usr/bin/python

# install dependencies
RUN wget https://bootstrap.pypa.io/get-pip.py && \
	python3 get-pip.py && \
	rm get-pip.py

RUN pip3 install PyYAML
RUN pip3 install matplotlib


RUN export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/cuda-10.2/lib64 
#RUN export CUDA_HOME = "/usr/local/cuda-10.2/"
ENV FORCE_CUDA="1"

RUN git clone https://github.com/NVIDIA-AI-IOT/torch2trt
RUN cd torch2trt && python3 setup.py install --plugins

```

I can build the image succesfully without torch2trt, and also install torch2trt inside the docker, but it not persist.

Thanks in advance!

**Comment by jaybdub** [comment]
Hey @simaiden ,

Thanks for reaching out!

My guess is that the ``nvidia-runtime`` is not enabled by default.  This typically explains CUDA related commands succeed at runtime (when using ``--runtime nvidia``) but fail during docker build.

You can enable NVIDIA runtime at build time by default by edting ``/etc/docker/daemon.json`` as explained here https://docs.nvidia.com/dgx/nvidia-container-runtime-upgrade/index.html#using-nv-container-runtime

In summary,

```bash
sudo vim /etc/docker/daemon.json
```

Set the default runtime

```json
{
 "default-runtime": "nvidia",
... 
}
```

Then restart docker

```bash
sudo systemctl restart docker
```

I think this should work after that point.

Please let me know if this works for you or you run into any issues.

Best,
John

**Comment by simaiden** [comment]
> Hey @simaiden ,
> 
> Thanks for reaching out!
> 
> My guess is that the `nvidia-runtime` is not enabled by default. This typically explains CUDA related commands succeed at runtime (when using `--runtime nvidia`) but fail during docker build.
> 
> You can enable NVIDIA runtime at build time by default by edting `/etc/docker/daemon.json` as explained here https://docs.nvidia.com/dgx/nvidia-container-runtime-upgrade/index.html#using-nv-container-runtime
> 
> In summary,
> 
> ```shell
> sudo vim /etc/docker/daemon.json
> ```
> 
> Set the default runtime
> 
> ```json
> {
>  "default-runtime": "nvidia",
> ... 
> }
> ```
> 
> Then restart docker
> 
> ```shell
> sudo systemctl restart docker
> ```
> 
> I think this should work after that point.
> 
> Please let me know if this works for you or you run into any issues.
> 
> Best,
> John

Thanks John! It works!
